test/object_store/test_backup.py

```diff
@@ -106,6 +106,42 @@ async def test_backup_to_non_existent_bucket(manager: ManagerClient, s3_server):
     assert status['state'] == 'failed'
 
 
+@pytest.mark.asyncio
+async def test_backup_to_non_existent_snapshot(manager: ManagerClient, s3_server):
+    '''backup should fail if the snapshot does not exist'''
+
+    cfg = {'enable_user_defined_functions': False,
+           'object_storage_config_file': str(s3_server.config_file),
+           'experimental_features': ['keyspace-storage-options'],
+           'task_ttl_in_seconds': 300
+           }
+    cmd = ['--logger-log-level', 'snapshots=trace:task_manager=trace']
+    server = await manager.server_add(config=cfg, cmdline=cmd)
+    ks, cf = await prepare_snapshot_for_backup(manager, server)
+
+    workdir = await manager.server_get_workdir(server.server_id)
+    cf_dir = os.listdir(f'{workdir}/data/{ks}')[0]
+    files = set(os.listdir(f'{workdir}/data/{ks}/{cf_dir}/snapshots/backup'))
+    assert len(files) > 0
```

Comment:
Are the above checks necessary? Backup is not supposed to work anyway

Reply:
i guess not, this behavior should have been covered by `test_simple_backup`. will drop it. and, by "checks" you meant "check", right? there is only a single `assert` before your comment.